### PR TITLE
feat: During building block creation, add the 'sap.fe.macros' library to the 'manifest.json' if it is not already listed

### PR DESCRIPTION
**Pull request by 815are**
All changes related to creation of building blocks like `Chart`, `Table`, `Filterbar`.
Changes:
1. If 'sap.fe.templates' or 'sap.fe.core' are not listed in `manifest.json`, then do not throw error about missing dependency;
2. If 'sap.fe.macros' is not listed in `manifest.json`, then automatically add it to 'manifest.json' during building block creation;

### CODE REVIEW

**Review by AlinaGovoruhina — APPROVED** [review]
Code is clear and understandable, unit test coverage good, changeset present

File: packages/fe-fpm-writer/src/building-block/index.ts [hunk 1]
```diff
@@ -7,12 +7,13 @@ import { BuildingBlockType, type BuildingBlock, type BuildingBlockConfig, type B
 import { DOMParser, XMLSerializer } from '@xmldom/xmldom';
 import * as xpath from 'xpath';
 import format from 'xml-formatter';
-import { getErrorMessage, validateBasePath } from '../common/validate';
+import { getErrorMessage, validateBasePath, validateDependenciesLibs } from '../common/validate';
 import { getTemplatePath } from '../templates';
 import { CodeSnippetLanguage, type FilePathProps, type CodeSnippet } from '../prompts/types';
 import { coerce, lt } from 'semver';
 import type { Manifest } from '../common/types';
 import { getMinimumUI5Version } from '@sap-ux/project-access';
```
Comment (devinea) [inline packages/fe-fpm-writer/src/building-block/index.ts:15]:
```suggestion
import { getMinimumUI5Version, getWebappPath } from '@sap-ux/project-access';
```

File: packages/fe-fpm-writer/src/building-block/index.ts [hunk 2]
```diff
@@ -25,6 +26,16 @@ interface MetadataPath {
...
 }
 
+/**
+ * Gets manifest path.
+ *
+ * @param {string} basePath the base path
+ * @returns {Manifest | undefined} path to manifest file
+ */
+function getManifestPath(basePath: string): string {
+    return join(basePath, 'webapp/manifest.json');
```
Comment (devinea) [inline packages/fe-fpm-writer/src/building-block/index.ts:36]:
```suggestion
    return join(await getWebappPath(basePath), FileName.Manifest);
```
Better not to hardcode webapp path

**Review by devinea — COMMENTED** [review]
Code changes are clear and covered by tests
changset ✅ 

1 small suggestion

File: packages/fe-fpm-writer/src/building-block/index.ts (hunk 2 above)
Comment (815are) [reply packages/fe-fpm-writer/src/building-block/index.ts]: @devinea 
applied correct, but one of problem is that API methods became asynchronous -> commit ccfa3e095b334bc5f70df2048b5bb2caaaf830f9

**Review by devinea — APPROVED** [review]
Thanks for the updates. Re-approving.

**Review by devinea — APPROVED** [review]

**Review by AlinaGovoruhina — APPROVED** [review]
Reapproving methods changed to async

**Pull request merged**